contracts/RecruitmentV2.sol

```diff
@@ -0,0 +1,434 @@
+            candidates[_candidateEmail] = Candidate(
+                msg.sender,
+                _candidateEmail,
+                0
+            );
+        }
+
+        referrals[_referralId].isConfirmed = true;
+        emit ReferralConfirm(_referralId, _jobId, _candidateEmail);
+    }
+
+    /// Hire a candidate
+    /// @param referralId referral id of the referral
+    function hireCandidate(
+        uint256 referralId
+    ) external onlyRegisteredCompany onlyOpenJob(referrals[referralId].jobId) {
+        Referral storage referral = referrals[referralId];
+        Job storage job = jobs[referral.jobId];
+
+        require(msg.sender == job.company, "Only the company can hire");
+        require(!referral.isHired, "Candidate already hired");
+
+        referral.isHired = true;
+        job.isOpen = false;
+        job.hiredReferralId = referralId;
+        companyHiredCandidate[msg.sender][referral.candidateEmail] = true;
+    }
+
+    /// returns the jobs created by a company
+    /// @param _company address of the company to retrive the jobs
+    function getCompanyJobs(
+        address _company
+    ) public view returns (uint256[] memory) {
+        require(
+            companies[_company].companyAddress != address(0),
+            "Company is not registered"
+        );
+        return companyToJobs[_company];
+    }
+
+    /// returns the referrals made by a referrer
+    /// @param referrer address of the referrer
+    function getReferrerReferrals(
+        address referrer
+    ) public view returns (uint256[] memory) {
+        require(
+            referrers[referrer].reffererAddress != address(0),
+            "Referrer is not registered"
+        );
+        return referrerToReferrals[referrer];
+    }
+
+    /// Close a Job without hiring
+    /// @param jobId job id of the job
+    function closeJob(
+        uint256 jobId
+    ) external onlyRegisteredCompany onlyOpenJob(jobId) {
+        Job storage job = jobs[jobId];
+        require(
+            msg.sender == job.company,
+            "Only the company can close the job"
+        );
+        job.isOpen = false;
+        token.transfer(job.company, job.bounty); // transfer the bounty back to the company
+    }
+
+    /// Dirburse Job bounty after 90 days
+    /// @param _jobId job id of the job for which the bounty is to be disbursed
+    function disburseBounty(
+        uint256 _jobId
+    ) external nonReentrant onlyRegisteredCompany {
+        require(
+            jobs[_jobId].company == msg.sender,
+            "Only the company can disburse bounty"
+        );
+        require(jobs[_jobId].isOpen == false, "Job is still open");
+        require(jobs[_jobId].isDisbursed == false, "Bounty already disbursed");
+        require(
+            jobs[_jobId].creationTime + 90 days < block.timestamp,
+            "90 days are not completed yet"
+        );
+        uint256 bounty = jobs[_jobId].bounty;
+        uint256 referralId = jobs[_jobId].hiredReferralId;
```

Comment:
need to handle if a job is closed and no hiredReferrals

Reply:
if the company closes a job without hiring there will not be any bounty to disburse so it's a different flow.